src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs

```diff
@@ -506,5 +506,43 @@ public async Task<IActionResult> EditPassword(ResetPasswordViewModel model)
 
             return View(model);
         }
+
+        [HttpPost]
+        public async Task<IActionResult> Unlock(string id)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageUsers))
+            {
+                return Forbid();
+            }
+
+            var user = await _userManager.FindByIdAsync(id) as User;
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            await _userManager.ResetAccessFailedCountAsync(user);
+            var result = await _userManager.SetLockoutEnabledAsync(user, false);
+
+            if (result.Succeeded)
+            {
+                await _userManager.SetLockoutEndDateAsync(user, null);
```

Comment:
Should it also call `ResetAccessFailedCountAsync(0)` ?

Reply:
`ResetAccessFailedCountAsync` is called 5 lines above.